Vialer/AppDelegate.m

```diff
@@ -186,7 +175,7 @@ - (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(
 
 #pragma mark - VoiP push notifications
 - (void)registerForVoIPNotifications {
-    [self.pzPushHandlerMiddleware registerForVoIPNotifications];
+    [[PZPushMiddleware sharedInstance]  registerForVoIPNotifications];
```

Comment:
I see no reason why this should be a singleton. Only the AppDelegate uses this middleware. So why not use a property for the middleware object?